contracts/root/staking/SupernetManager.sol

```diff
@@ -20,6 +20,7 @@ abstract contract SupernetManager is ISupernetManager, Initializable {
     }
 
     function onInit(uint256 id_) external onlyStakeManager {
+        require(id == 0, "SupernetManager: ID_ALREADY_SET");
```

Comment:
+1

Reply:
But even in that scenario, a SupernetManager should either be a) upgradeable b) allow such that it can only be set once - the StakeManager should not be able to update the value more than once basically.

Reply:
I see @QEDK 's point, @gretzke has recommended elsewhere putting the check in both, sounds like the most reasonable path

Reply:
Thanks, closing.